crates/air-lib/trace-handler/src/state_automata/par_fsm/state_handler/new_states_calculation.rs

```diff
@@ -32,20 +34,20 @@ pub(super) fn compute_new_states(
 
 fn compute_new_state(par_result: ParResult, subgraph_type: SubgraphType, slider: &TraceSlider) -> FSMResult<CtxState> {
     let par_subgraph_len = match subgraph_type {
-        SubgraphType::Left => par_result.left_size as usize,
+        SubgraphType::Left => par_result.left_size,
         SubgraphType::Right => par_result.size().ok_or(StateFSMError::ParLenOverflow(par_result))?,
     };
 
     let new_position = slider
         .position()
-        .checked_add(par_subgraph_len)
+        .checked_add(&air_interpreter_data::TracePos::from(par_subgraph_len))
         .ok_or_else(|| StateFSMError::ParPosOverflow(par_result, slider.position(), MergeCtxType::Previous))?;
 
     let new_subtrace_len = match subgraph_type {
-        SubgraphType::Left => par_subgraph_len,
+        SubgraphType::Left => par_subgraph_len as usize,
```

Comment:
Is it always valid conversion? You mentioned something in the daily meet.

Reply:
We are converting from presumably lower type to either bigger or equal in terms of values range.

Reply:
One may add a static assert (from `static_assertions` crate or `static_assert_macro`) that checks that presumption on compile time (and also makes this assumption explicit).  Or use `try_into` that should be optimized out if the conversion is really infallible.

Reply:
Thx for the clarification.